R/CohortConstruction.R

```diff
@@ -106,24 +106,28 @@ generateCohortSet <- function(connectionDetails = NULL,
   if (incremental) {
     recordKeepingFile <- file.path(incrementalFolder, "GeneratedCohorts.csv")
 
-    if (isTRUE(attr(cohortDefinitionSet, "hasSubsetDefinitions"))) {
-      cohortDefinitionSet$checksum <- ""
-      for (i in 1:nrow(cohortDefinitionSet)) {
+    cohortDefinitionSet$checksum <- ""
+    for (i in 1:nrow(cohortDefinitionSet)) {
+      if (isTRUE(attr(cohortDefinitionSet, "hasSubsetDefinitions"))) {
         # This implementation supports recursive definitions (subsetting subsets) because the subsets have to be added in order
-        if (cohortDefinitionSet$subsetParent[i] != cohortDefinitionSet$cohortId[i]) {
+        if (cohortDefinitionSet$isSubset[i] && cohortDefinitionSet$subsetParent[i] != cohortDefinitionSet$cohortId[i]) {
           j <- which(cohortDefinitionSet$cohortId == cohortDefinitionSet$subsetParent[i])
           cohortDefinitionSet$checksum[i] <- computeChecksum(paste(
             cohortDefinitionSet$sql[j],
             cohortDefinitionSet$sql[i]
           ))
-        } else {
-          cohortDefinitionSet$checksum[i] <- computeChecksum(cohortDefinitionSet$sql[i])
         }
+      } else if (isTRUE(attr(cohortDefinitionSet, "hasCombinedCohorts"))) {
+        dependantCohortIds <- as.integer(strsplit(cohortDefinitionSet$dependentCohorts[i]))
+        dependentCohortIdx <- which(cohortDefinitionSet$cohortId %in% dependantCohortIds)
+        cohortDefinitionSet$checksum[i] <- 
```

Comment:
The loop logic is now:  if it is a subset cohort, calc checksum one way, if it is combined cohort, do it another way, else do it the simple 'by sql' way.